### PR TITLE
[incubator/kafka] Allow configurationOverride of zookeeper.connect

### PR DESCRIPTION
**Pull request by ghost**
This PR allows use of `zookeeper.connect` at `configurationOverrides`.  It is backwards compatible with the zookeeper url+port method of setting `zookeeper.connect`.  This override is believed necessary because the Zookeeper connect string is more complex than just url and port.

### CODE REVIEW

**Comment by ghost** [comment]
/assign @viglesiasce

cc @benjigoldberg

File: incubator/kafka/templates/tests/test_topic_create_consume_produce.yaml [hunk 1]
```diff
@@ -13,7 +13,13 @@ spec:
     - -c
     - |
       # Create the topic
-      kafka-topics --zookeeper {{ template "zookeeper.url" . }} --topic helm-test-topic-create-consume-produce --create --partitions 1 --replication-factor 1 --if-not-exists && \
+      kafka-topics --zookeeper \
+{{- if not ( index .Values "configurationOverrides" "zookeeper.connect" ) }}
```
Comment (benjigoldberg) [inline incubator/kafka/templates/tests/test_topic_create_consume_produce.yaml]: @josdotso you could wrap this conditional logic into the zookeeper helper template to DRY it up throughout this chart. What do you think?

Comment (ghost) [reply incubator/kafka/templates/tests/test_topic_create_consume_produce.yaml]: Good idea @benjigoldberg . Please let me know if the latest commit is agreeable.

**Comment by benjigoldberg** [comment]
@josdotso looks great! LGTM

**Comment by viglesiasce** [comment]
@benjigoldberg would you mind if I added you to the OWNERS file so that you can merge these PRs on your own moving forward?

**Comment by benjigoldberg** [comment]
@viglesiasce No problem! Would love to help out.

**Pull request merged**